.github/workflows/scripts/pr_check.py

```
@@ -0,0 +1,224 @@
+    logger.info(f"Target branch '{my_br.name}' sha: {br_sha}")
+
+    # Apply the cherry-picks
+    for com in commits:
+        # Get the commit object for the current commit SHA
+        commit = repo.get_commit(com)
+
+        # Retrieve the GitTree object using the commit's tree SHA
+        tree = repo.get_git_tree(commit.commit.tree.sha)
+
+        # Create a new commit with the same message and tree, but with the current branch's latest commit as the parent
+        try:
+            new_commit = repo.create_git_commit(
+                message=commit.commit.message,
+                tree=tree,
+                parents=[repo.get_commit(br_sha).commit]
+            )
+        except:
+            logger.error(f"Failed to create commit!")
+            sys.exit(1)
+
+        # Update the branch SHA to the new commit's SHA
+        br_sha = new_commit.sha
+        logger.debug(f"Cherry-picked commit {commit.sha} to {new_commit.sha}")
+
+    # Update the reference of the branch to the new commit
+    try:
+        ref = repo.get_git_ref(f"heads/{auto_branch}")
+    except:
+        logger.error(f"Failed to get git reference!")
+        sys.exit(1)
+    ref.edit(sha=br_sha)
+    logger.info(f"Updated branch '{auto_branch}' to new commit {br_sha}")
+
+
+def create_pull_request_if_needed(repo: Repository, auto_branch: str, target_br: str) -> None:
+    """Create a pull request if one does not already exist."""
+    pr_auto_title = f"[AUTO_UPDATE] Branch {target_br}"
+    prs = repo.get_pulls(state="open", base=target_br)
+    if not any(pr.title == pr_auto_title for pr in prs):
+        # Create a new pull request with the auto_update branch as the head and the target branch as the base
+        try:
+            repo.create_pull(
+                title=pr_auto_title,
+                body="Automated update",
+                head=auto_branch,
+                base=target_br
+            )
+        except:
+            logger.error(f"Failed to create PR!")
```

Error is not handled

---

Added an exit code. It will be caught by the calling workflow